### ISSUE TITLE
nan occur 

### ISSUE THREAD

**Issue by zh7117** [YunYang1994/tensorflow-yolov3#471]
during training
nan loss would appear randomly. usually it's test loss, and it might be normal after several epochs training.
but if training loss became nan, the back propagation steps would update all the weights to nan, so the loss would always be nan.
some tips to avoid nan:
1) add eps to denominator in iou and giou calculation 
2) add eps to to cross_entropy calculation
3) clip gradient before updating weights.
